src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -0,0 +1,121 @@
+
+        private string _name;
+
+        /// <summary>
+        /// Initializes a new instance of the ExtensibleEnum class.
+        /// </summary>
+        /// <param name="name">The value for this instance of the extensible enumeration.</param>
+        protected ExtensibleEnum(string name)
+        {
+            _name = name;
+        }
+
+        /// <summary>
+        /// Compares two ExtensibleEnum values for equality.
+        /// </summary>
+        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <returns>true if the ExtensibleEnum objects are equal or are both null; false otherwise.</returns>
+        public static bool operator ==(ExtensibleEnum<T> lhs, T rhs) => object.Equals(lhs, rhs);
+
+        /// <summary>
+        /// Compares two ExtensibleEnum values for inequality.
+        /// </summary>
+        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <returns>true if the ExtensibleEnum objects are not equal; false otherwise.</returns>
+        public static bool operator !=(ExtensibleEnum<T> lhs, T rhs) => !object.Equals(lhs, rhs);
+
+        /// <summary>
+        /// Defines explicit conversion from ExtensibleEnum to string.
+        /// </summary>
+        /// <param name="name">ExtensibleEnum to convert.</param>
+        /// <returns>The ExtensibleEnum as a string.</returns>
+        public static explicit operator string(ExtensibleEnum<T> name) => name?.ToString();
+
+        /// <summary>
+        /// Looks up an ExtensibleEnum instance by name, or returns null if the given name does not match one of the
+        /// known values of this ExtensibleEnum.
+        /// </summary>
+        /// <param name="name">Name of the ExtensibleEnum value.</param>
+        /// <returns>An instance of type T with the given name, or null if no such instance could be found.</returns>
+        protected static T Lookup(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(name, "name");
+            }
+
+            T enumValue;
+            return _nameMap.Value.TryGetValue(name, out enumValue) ? enumValue : null;
+        }
+
+        /// <summary>
+        /// Compares the ExtensibleEnum for equality with another ExtensibleEnum.
+        /// </summary>
+        /// <param name="other">The ExtensibleEnum with which to compare.</param>
+        /// <returns>true if the ExtensibleEnum objects are equal; false otherwise.</returns>
+        public bool Equals(T other)
+        {
+            if (object.ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return this._name == other._name;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
```

Comment:
What's the policy on C# 6 syntax in this repo? Can we simplify these to be expression-bodied methods (`=>`)?

Reply:
Nevermind, I see I already used `=>` above. I must have missed these.